src/server/services/wms/qgswmsrendercontext.cpp

```diff
@@ -638,22 +648,40 @@ bool QgsWmsRenderContext::isValidWidthHeight() const
   return true;
 }
 
+QgsRectangle QgsWmsRenderContext::mapExtent() const
+{
+  QgsRectangle extent = mParameters.bboxAsRectangle();
+  if ( !mParameters.bbox().isEmpty() && extent.isEmpty() )
+  {
+    throw QgsBadRequestException( QgsServiceException::QGIS_InvalidParameterValue,
+                                  mParameters[QgsWmsParameter::BBOX] );
+  }
+  const double extentWidth = extent.width();
```

Comment:
Is const really necessary here? and I would also recommend to remove the intermediate extentheight, extentWidth, width, height variables